src/components/OverviewMap.js

```diff
@@ -146,53 +92,78 @@ class OverviewMap extends React.Component {
-        const latLngs = geojson.coordinates.map(([lng, lat]) => new LatLng(lat, lng));
-        return (<Polyline key={Math.random()} positions={latLngs}>{this.getPopupContent(hearing, geojson)}</Polyline>);
-      }
-      case "Feature": {
-        /**
-         * Recursively get the Map element
-         */
-        return (this.getMapElement(geojson.geometry, hearing));
+  getMapElement(hearing) {
+    const {mapElementLimit} = this.props;
+    const {geojson, id} = hearing;
+    if (geojson) {
+      switch (geojson.type) {
+        case "Polygon": {
+          // XXX: This only supports the _first_ ring of coordinates in a Polygon
+          const latLngs = geojson.coordinates[0].map(([lng, lat]) => new LatLng(lat, lng));
+          return (
+            <Polygon
+              key={id + "" + Math.random()}
+              positions={latLngs}
+            >
+              {this.getPopupContent(hearing, geojson)}
+            </Polygon>);
+        }
+        case "Point": {
+          const latLngs = new LatLng(geojson.coordinates[1], geojson.coordinates[0]);
+          return (
+            <Marker
+              position={latLngs}
+              key={id + "" + Math.random()}
+              icon={new Leaflet.Icon({
+                iconUrl: leafletMarkerIconUrl,
+                shadowUrl: leafletMarkerShadowUrl,
+                iconRetinaUrl: leafletMarkerRetinaIconUrl,
+                iconSize: [25, 41],
+                iconAnchor: [13, 41]
+              })}
+              {...this.getAdditionalParams(hearing)}
+            >{this.getPopupContent(hearing, geojson)}
+            </Marker>
+          );
+        }
+        case "LineString": {
+          const latLngs = geojson.coordinates.map(([lng, lat]) => new LatLng(lat, lng));
+          return (
+            <Polyline key={id + "" + Math.random()} positions={latLngs}>
+              {this.getPopupContent(hearing, geojson)}
+            </Polyline>
+          );
+        }
+        case "FeatureCollection": {
+          // if mapElementLimit is true & more than 0, display up to that amount of elements
+          const mapElementArray = [];
+          if (mapElementLimit && mapElementLimit > 0) {
+            geojson.features.slice(0, mapElementLimit).forEach((feature) => {
+              mapElementArray.push(this.getMapElement(feature, hearing));
+            });
+          } else {
+            // if mapElementLimit is false -> display all elements found in FeatureCollection
+            geojson.features.forEach((feature) => {
+              mapElementArray.push(this.getMapElement(feature, hearing));
+            });
+            return mapElementArray;
+          }
+        }
+          break;
+        case "Feature": {
+          /**
+           * Recursively get the Map element
+           */
+          return (this.getMapElement(geojson.geometry, hearing));
+        }
+        default:
+          // TODO: Implement support for other geometries too (markers, square, circle)
```

Comment:
This doesnt work correctly for a hearing with multiple markers, did you test that this works correctly?  The refactored getMapElement only takes 1 param (hearing) and here we are passing 2 params to it. Here is an image of the geojson object of a hearing that has multiple markers(a FeatureCollection)
![image](https://user-images.githubusercontent.com/50320115/224994953-e39946b3-7fb7-4b61-abd2-c8325727109f.png)

Reply:
Yeah, I was worried something like this could happen. It seemed to work on my local tho as there was one test case with multiple point type notations and those were also correctly limited to 10 on home page view.

Reply:
Right, realized I had not rebuilt the thing and so it just seemed to still work, implemented a proper fix now @hienous, can you verify the fix works?